sql: reserve none and pg_* roles

otan opened the pull request:
Release note (sql change, backward-incompatible change): Roles with the
name `none` and starting with `pg_` can no longer be created. Any
existing roles with these names may continue to work, but they may be
broken when new features (e.g. SET ROLE) is introduced.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68972)
<!-- Reviewable:end -->

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/68972#-:-MhESd23A4DuFPGqHhgR:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan and @RichardJCai)

___
*[pkg/security/username.go, line 111 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/68972#-MhESQ3bE8BKTy8jxO5d:-MhESQ3bE8BKTy8jxO5e:b-j8c80t) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f78fcac6276f77c5153edbfb8f4d8dea1527777/pkg/security/username.go#L111)):*
> ```Go
> // Matches Postgres.
> func (s SQLUsername) IsReserved() bool {
> 	return s.IsPublicRole() || s.u == NoneRole || strings.HasPrefix(s.u, "pg_")
> ```

this is a totally speculative request but, can we reserve the `crdb_internal` prefix? I don't have any envisioned use case, I just don't see any good reason not to carve out a space of names that we could use if we so chose. 


<!-- Sent from Reviewable.io -->

otan committed:
sql: reserve none and pg_* roles

Release note (sql change, backward-incompatible change): Roles with the
name `none` and starting with `pg_` or `crdb_internal` can no longer be
created. Any existing roles with these names may continue to work, but
they may be broken when new features (e.g. SET ROLE) is introduced.

otan commented:
done, TFTR!

bors r=ajwerner
